docs/source/modeling.rst

```diff
@@ -1,10 +1,396 @@
+
+1. **Investment stage**: Estimates relevant investment costs such as
+   installation, expansion (which includes the installation of new
+   selected generation units), operating costs, penalty, and
+   maintenance costs. These costs include representative costs for
+   both thermal and renewable generators as well as transmission
+   equipment (e.g., transmission lines). Additionally, it declares a
+   discrete decision (represented as a disjunction through the GDP
+   formulation) to determine the operational state of the generation
+   and transmission equipment (operational, retired, etc.) that can
+   potentially influence the investment costs.
+
+2. **Commitment stage**: Determines the operational status of the
+   generators (set as a disjunction) while considering relevant
+   operational costs and constraints to ensure a reliable power
+   generation service. Additionally, it establishes operating limits
+   considering the availability of the equipment from the dispatch
+   stage.
+
+3. **Dispatch stage**: Defines the status of transmission lines using
+   a discrete decision (set as a disjunction) while considering
+   operational costs, operability limits (such as power flow, ramping,
+   reserves, etc.), and constraints to ensure the effective use of the
+   active power lines.
+
+Additionally, we include a data allocation step (prior to the
+formulation of the three stages) that assigns all relevant operating
+and cost parameters of the model. Since we are considering this as a
+data preprocessing step, not a stage in the formulation, this step is
+not represented in Figure 1.
+
+Below we outline the steps we followed to construct the GTEP
+model. Note that the first four steps correspond directly to the
+stages described above (in some steps, the description is accompanied
+by example code).
+
+
+**Step 1**: Data allocation
+
+(a) The initial step to create this model is to import the necessary
+    libraries.
+   
+    .. code-block::
+
+       import json
+       import numpy as np
+       import math
+       from math import ceil
+   
+       # Import Pyomo components
+       from pyomo.environ import * 
+       from pyomo.environ import units as u
+       from pyomo.common.timing import TicTocTimer
+       from pyomo.repn.linear import LinearRepnVisitor
+   
+       from egret.data.model_data import ModelData
+       from egret.model_library.transmission.tx_utils import scale_ModelData_to_pu
+       from config_options import _get_model_config
+
+       # Import IDAES-GTEP functions 
+       from gtep.gtep_model import model_data_references
+       from gtep.gtep_model import model_create_investment_stages
+       from gtep.gtep_model import create_objective_function
+
+(b) Continue the creation of the GTEP model by including a new Python
+    `class`, which includes a set of relevant parameters for the
+    solution of the model, as arguments of the function.
+
+    .. code-block::
+
+       class ExpansionPlanningModel:
```

Comment:
So is a user expected to _define_ this class every time or to create an _instance_ of this class?